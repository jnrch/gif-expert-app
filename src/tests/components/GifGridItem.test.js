import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Test in <GifGridItem />', () => {

    const title = "this is a gif webpage";
    const url = "https://this.com.ar";
    const wrapper = shallow( <GifGridItem title={title} url={url} />)
    
    test('should show component correctly', () => {

        expect( wrapper ).toMatchSnapshot();
        
    })

    test('should have a paragraph with title', () => {
        
        const p = wrapper.find('p');
        
        expect(p.text().trim()).toBe(title);
    })
    
    test('should have image equals to url and alt of the props', () => {

        const img = wrapper.find('img');

        expect( img.prop('src')).toBe(url)
        
    })

    test('should have animate__bounce class', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');

       expect(className.includes('animate__bounce')).toBe(true);
        
    })
    
    
    
})
