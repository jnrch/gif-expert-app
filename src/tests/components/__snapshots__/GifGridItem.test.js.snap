// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem /> should show component correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="this is a gif webpage"
    src="https://this.com.ar"
  />
  <p>
     
    this is a gif webpage
     
  </p>
</div>
`;
